src/backends/llvm.ts

```diff
@@ -1,50 +1,92 @@
 import * as ast from '../ast';
-import { Emitter, emit } from './emitter';
-import { varsym, is_fun_type } from './emitutil';
 import { ASTVisit, ast_visit, complete_visit } from '../visit';
 import { INT, FLOAT, Type, OverloadedType, FunType } from '../type';
+import { Proc, Prog, Variant, CompilerIR } from '../compile/ir'
 import * as llvm from '../../node_modules/llvmc/src/wrapped';
-import { Proc, Prog, CompilerIR, Variant } from '../compile/ir';
+
+///////////////////////////////////////////////////////////////////
+// Begin Emit Functions & Redundant Funcs
+///////////////////////////////////////////////////////////////////
 
 /**
  * Like `emitter.Emitter`, but for generating LLVM code instead of strings.
  */
 interface LLVMEmitter {
-  ir: CompilerIR;
-
   /**
    * The LLVM IRBuilder object used to generate code.
    */
   builder: llvm.Builder;
 
+  /**
+   * Map from id's to Alloca ptr's
+   */
+  namedValues: {[id:string] : llvm.Value};
```

Comment:
This "symbol table" is currently indexed by string. That is, if you have a variable defined at node 42, you currently need to do `namedValues[varsym(42)]`. I think we can get away with indexing this *directly* on the node id (i.e., `tree.id!`, which is an integer). Then this can just have the type `llvm.Value[]` because JavaScript arrays can be sparse.